src/lib/ai/summary.ts

```diff
@@ -53,6 +54,18 @@ class Summary extends Cache {
       throw e.error?.code ? new Error(e.error?.code.toUpperCase()) : e;
     }
   };
+
+  #cacheExpired = () => {
+    const { id, state, updated } = this.proposal!;
+
+    if (state !== 'pending') return false;
+
+    return tempCacheIds.has(id) && tempCacheIds.get(id) !== updated;
+  };
+
+  afterCreateCache() {
+    tempCacheIds.set(this.proposal!.id, this.proposal!.updated);
+  }
```

Comment:
I think we can't relay on updated param for cache, a pending proposal can be updated any number of times

Reply:
The `update` value will be cached only after cache creation. So each time it changes, it will invalidate the cache

Reply:
oh updated is timestamp. got it